plugins/builds/index.js

```diff
@@ -7,6 +7,57 @@ const stepGetRoute = require('./steps/get');
 const stepUpdateRoute = require('./steps/update');
 const stepLogsRoute = require('./steps/logs');
 const listSecretsRoute = require('./listSecrets');
+const workflowParser = require('screwdriver-workflow-parser');
+
+/**
+ * Start the build
+ * @method startBuild
+ * @param  {Object}   config                Configuration object
+ * @param  {Factory}  config.jobFactory     Job Factory
+ * @param  {Factory}  config.buildFactory   Build Factory
+ * @param  {Number}   config.pipelineId     Pipeline Id
+ * @param  {String}   config.jobName        Job name
+ * @param  {String}   config.username       Username of build
+ * @param  {String}   config.scmContext     SCM context
+ * @param  {Build}    config.build          Build object
+ * @return {Promise}
+ */
+function startBuild(config) {
+    const { jobFactory, buildFactory, pipelineId, jobName, username, scmContext, build } = config;
+
+    return jobFactory.get({
+        name: jobName,
+        pipelineId
+    }).then((job) => {
+        if (job.state === 'ENABLED') {
+            return buildFactory.create({
+                jobId: job.id,
+                sha: build.sha,
+                parentBuildId: build.id,
```

Comment:
Is `parentBuildId` needed?

Reply:
Nvm I see it in the database.
We should update the docs here at some point: https://github.com/screwdriver-cd/models/blob/master/lib/buildFactory.js#L145-L156

Reply:
Yep. parentBuildId is the build that triggers this one.